frontend/src/components/GAlert.vue

```diff
@@ -5,69 +5,101 @@ SPDX-License-Identifier: Apache-2.0
-    <div class="subtitle-1">
-      {{message}}
-      <v-btn dark outlined small v-if="!!detailedMessage" @click="detailedMessageVisible = !detailedMessageVisible">
-        Details
-      </v-btn>
-    </div>
-    <transition name="fade">
-      <div v-if="!!detailedMessageVisible">
-        <code>{{detailedMessage}}</code>
-      </div>
-    </transition>
+  <v-alert
+    class="alertBanner"
+    :type="type"
+    v-model="alertVisible"
+    :color="color"
+    :transition="transition"
+    @input="closeBanner"
+    dismissible
+  >
+    <div v-if="message" class="alert-banner-message" v-html="messageHtml"></div>
+    <slot v-else name="message"></slot>
   </v-alert>
 </template>
 
 <script>
+import { mapActions } from 'vuex'
+import { transformHtml } from '@/utils'
+
+const LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES = 'global/alert-banner/hidden-messages'
+
 export default {
-  name: 'galert',
   props: {
-    message: {
+    message: { // alternatively, use message slot
       type: String
     },
-    detailedMessage: {
+    type: {
+      type: String,
+      default: 'error'
+    },
+    identifier: { // pass identifier to permanently hide the message on close
       type: String
     },
     color: {
-      type: String,
-      required: true
-    },
-    tile: {
-      type: Boolean
+      type: String
     },
-    dark: {
-      type: Boolean,
-      default: true
+    transition: {
+      type: String
     }
   },
   data () {
     return {
-      detailedMessageVisible: false
+      alertVisible: false
     }
   },
   computed: {
-    alertVisible: {
-      get () {
-        return !!this.message
-      },
-      set (value) {
-        if (!value) {
-          this.$emit('update:message', undefined)
-          this.$emit('update:detailedMessage', undefined)
-        }
+    messageHtml () {
+      return transformHtml(this.message)
+    }
+  },
+  methods: {
+    ...mapActions([
+      'setAlertBanner'
+    ]),
+    async closeBanner () {
+      if (this.identifier) { // hide permanently
+        this.hidePermanently(this.identifier)
       }
+
+      this.setAlertVisibility(false)
+    },
+    hidePermanently (identifier) {
+      const permanentlyHiddenIds = this.$localStorage.getObject(LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES) || {}
+      permanentlyHiddenIds[identifier] = true
+      this.$localStorage.setObject(LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES, permanentlyHiddenIds)
+    },
+    isPermanentlyHidden (identifier) {
+      if (!identifier) {
+        return false
+      }
+
+      const permanentlyHiddenIds = this.$localStorage.getObject(LOCAL_STORE_ALERT_BANNER_HIDDEN_MESSAGES) || {}
+      return permanentlyHiddenIds[this.identifier] === true
+    },
+    updateAlertVisibility () {
+      const visible = !this.isPermanentlyHidden(this.identifier)
+      this.setAlertVisibility(visible)
+    },
+    setAlertVisibility (visible) {
+      this.alertVisible = visible
+      this.$emit('value', visible)
```

Comment:
Why do we emit visibility?
Why is the name value? Shouldn't it be a value prop with two ways data binding and emit `input`?

Reply:
Either add a value (v-model) property and emit input or just emit input with value prop.

Reply:
I removed the emit completely. The component handles the visibility which depends if the user already dismissed the message already or not